src/go_caw/main.go

```diff
@@ -0,0 +1,237 @@
+	flag_user         = flag.String("user", "", "Logs in as the given username")
+	flag_caw          = flag.String("caw", "", "Creates a new caw with the given text")
+	flag_reply        = flag.String("reply", "", "Indicates that the new caw is a reply to the given id")
+	flag_follow       = flag.String("follow", "", "Starts following the given username")
+	flag_read         = flag.String("read", "", "Reads the caw thread starting at the given id")
+	flag_profile      = flag.Bool("profile", false, "Gets the user’s profile of following and followers")
+)
+
+// NOTE: these do not match the C++ Caw implementation order for enum in event_type.h
+// and enum CommandInput in caw.cc, because this is a "mixed" enumeration so that
+// there is no need for two separate enums in the same program.
+type EventType int
+
+const (
+	kRegisteruser EventType = iota
+	kCaw
+	kFollow
+	kRead
+	kProfile
+	kInvalidCommand
+	kHookAll
+	kUnhookAll
+	kReply
+)
+
+// extracts and validates command provided by global flags
+func getCommand() EventType {
+	var hook, unhook, registeruser, user, caw, reply, follow, read, profile bool = *flag_hookall, *flag_unhookall, *flag_registeruser != "", *flag_user != "", *flag_caw != "",
+		*flag_reply != "", *flag_follow != "", *flag_read != "", *flag_profile
+	if !registeruser && !user && !caw && !reply && !follow && !read &&
+		!profile {
+		if hook && !unhook {
+			return kHookAll // ./caw --hookall
+		} else if !hook && unhook {
+			return kUnhookAll // ./caw --unhookall
+		}
+	} else if !hook && !unhook {
+		if registeruser && !user && !caw && !reply && !follow && !read &&
+			!profile {
+			return kRegisteruser // ./caw --registeruser <username>
+		} else if !registeruser && user {
+			if caw && !follow && !read && !profile {
+				if caw && !reply {
+					return kCaw // ./caw --user <username> --caw <caw text>
+				} else if caw && reply {
+					return kReply // ./caw --user <username> --caw <caw text>
+					// --reply <reply caw id>
+				}
+			} else if !caw && follow && !read && !profile {
+				return kFollow // ./caw --user <username> --follow <username>
+			} else if !caw && !follow && read && !profile {
+				return kRead // ./caw --user <username> --read <caw id>
+			} else if !caw && !follow && !read && profile {
+				return kProfile // ./caw --user <username> --profile
+			}
+		}
+	}
+	return kInvalidCommand
+}
+
+// gets event request through serializing and packing payload
+func getEventRequest(request proto.Message, event EventType) *faz.EventRequest {
+	serialized_payload, err := ptypes.MarshalAny(request)
+	if err != nil {
+		panic(err)
+	}
+	event_request := &faz.EventRequest{
+		EventType: int32(event),
+		Payload:   serialized_payload,
+	}
+	return event_request
+}
+
+func main() {
+	flag.Parse()
+
+	// GRPC
+	var conn *grpc.ClientConn
+	conn, err := grpc.Dial("0.0.0.0:50000", grpc.WithInsecure())
+	if err != nil {
+		fmt.Println("sorry, could not connect to Faz")
```

Comment:
Maybe look into fmt.Errorf() for similar error messages?

Reply:
I was using fmt error earlier, but I was running into weird issues for some of my functions.  Let me try it again in my fix commit and see if it works

Reply:
yeah I couldn't seem to fix some of the odd formatting, I will keep trying but might just have to leave it

Reply:
Yeah, I don't think it's a really major issue. fmt.Println() is fine.